File: Personal/Solvers/phone_solver.py
import sys
import random
import time

def how_long(max_time):
    my_time = random.randint(1, max_time)
    sis_time = random.randint(1, max_time)
    cur_time = max(my_time, sis_time)
    while my_time != sis_time:
        if cur_time > my_time:
            my_time += random.randint(1, max_time)
        else:
            sis_time += random.randint(1, max_time)
        cur_time = max(my_time, sis_time)
    return my_time

def run(num_sim, max_time):
    total_time = 0
    for i in range(num_sim):
        total_time += how_long(max_time)
    return float(total_time) / num_sim

num_sim = input("Number of Simulations: ")
max_time = input("Maximum Time: ")
run_time = time.clock();
average_time = run(num_sim, max_time)
print "Average time: " + str(round(100 * average_time) / 100)
print time.clock() - run_time;

File: Personal/Solvers/PhoneSolver.java
/**
 * Simulates the problem found here: 
 * http://fivethirtyeight.com/features/how-long-will-your-smartphone-distract-you-from-family-dinner/
 *
 * @author John Maxwell Gale Carnahan IV
 * @version 0.9.0.
 */

import java.util.Random;
import java.util.Scanner;

public class PhoneSolver {
    private static int numSim, maxTime;
    private static Random rand;

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Number of Simulations: ");
        numSim = scan.nextInt();
        System.out.println("Maximum Time: ");
        maxTime = scan.nextInt();

        rand = new Random();
        long runTime = System.currentTimeMillis();
        int totalTime;
        totalTime = 0;
        for (int i = 0; i < numSim; i++) {
            totalTime += simulateDay();
        }

        double averageTime;
        averageTime = (double) totalTime / numSim;
        System.out.printf("Average Time: %.3f \n", averageTime);

        double runDuration; 
        runDuration = (double)(System.currentTimeMillis() - runTime) / 1000;
        System.out.printf("Runtime: %.4f", runDuration);
    }

    private static int simulateDay() {
        Random rand = new Random();
        int myTime, sisTime, curTime;
        myTime = rand.nextInt(maxTime) + 1;
        sisTime = rand.nextInt(maxTime) + 1;
        curTime = Math.max(myTime, sisTime);
        while (myTime != sisTime) {
            if (myTime < curTime) {
                myTime += rand.nextInt(maxTime) + 1;
            } else {
                sisTime += rand.nextInt(maxTime) + 1;
            }
            curTime = Math.max(myTime, sisTime);
        }
        return myTime;
    } 
}